libtiledbvcf/src/stats/allele_count.cc

```diff
@@ -317,17 +327,38 @@ void AlleleCount::flush() {
         .set_offsets_buffer(COLUMN_NAME[GT], gt_offsets_)
         .set_data_buffer(COLUMN_NAME[COUNT], count_buffer_);
 
-    auto st = query_->submit();
+    query_->submit();
 
-    if (st != Query::Status::COMPLETE) {
-      LOG_FATAL("AlleleCount: error submitting TileDB write query");
+    if (remote_) {
+      if (query_->query_status() == Query::Status::FAILED) {
+        LOG_FATAL("AlleleCount: error submitting TileDB write query");
+      }
+    } else {
+      if (query_->query_status() != Query::Status::COMPLETE) {
+        LOG_FATAL("AlleleCount: error submitting TileDB write query");
+      }
     }
 
     // Insert sample names from this query into the set of fragment sample names
     fragment_sample_names_.insert(sample_names_.begin(), sample_names_.end());
     sample_names_.clear();
   }
 
+  // For remote global order writes, zero query buffers prior to
+  // submit_and_finalize.
+  if (remote_) {
+    query_
+        ->set_buffer(
+            COLUMN_NAME[CONTIG], contig_offsets_.data(), 0, &contig_buffer_, 0)
+        .set_buffer(COLUMN_NAME[POS], pos_buffer_.data(), 0)
+        .set_buffer(COLUMN_NAME[REF], ref_offsets_.data(), 0, &ref_buffer_, 0)
+        .set_buffer(COLUMN_NAME[ALT], alt_offsets_.data(), 0, &alt_buffer_, 0)
+        .set_buffer(
+            COLUMN_NAME[FILTER], filter_offsets_.data(), 0, &filter_buffer_, 0)
+        .set_buffer(COLUMN_NAME[GT], gt_offsets_.data(), 0, &gt_buffer_, 0)
+        .set_buffer(COLUMN_NAME[COUNT], count_buffer_.data(), 0);
```

Comment:
These buffers are cleared by the `clear` statements below. We can reuse the `set_data_buffer` /`set_offsets_buffer` code above after the buffers are cleared, and it doesn't need to depend on `remote_`.